hail/python/hail/expr/__init__.py

```diff
@@ -7,14 +7,43 @@
 from .expressions import eval, eval_typed, eval_timed, \
+    Indices, Aggregation, apply_expr, construct_expr, construct_variable, \
+    construct_reference, impute_type, to_expr, cast_expr, unify_all, \
+    unify_types_limited, unify_types, unify_exprs, \
     Expression, ExpressionException, ArrayExpression, ArrayNumericExpression, \
     BooleanExpression, CallExpression, CollectionExpression, DictExpression, \
     IntervalExpression, LocusExpression, NumericExpression, Int32Expression, \
     Int64Expression, Float32Expression, Float64Expression, SetExpression, \
-    StringExpression,  StructExpression, TupleExpression, NDArrayExpression, \
-    NDArrayNumericExpression
-from .functions import *
-from .functions import _showstr, _sort_by, _compare, _values_similar, _locus_windows_per_contig
+    StringExpression, StructExpression, TupleExpression, NDArrayExpression, \
+    NDArrayNumericExpression, \
+    expr_any, expr_int32, expr_int64, \
+    expr_float32, expr_float64, expr_call, expr_bool, expr_str, expr_locus, \
+    expr_interval, expr_array, expr_ndarray, expr_set, expr_dict, expr_tuple, \
+    expr_struct, expr_oneof, expr_numeric, coercer_from_dtype
+from .functions import literal, chi_squared_test, if_else, cond, switch, case, \
+    bind, rbind, contingency_table_test, dbeta, dict, dpois, exp, entropy, \
+    fisher_exact_test, gp_dosage, hardy_weinberg_test, parse_locus, \
+    parse_variant, variant_str, locus, locus_from_global_position, interval, \
+    locus_interval, parse_locus_interval, call, is_defined, is_missing, \
+    is_nan, is_finite, is_infinite, json, parse_json, log, log10, null, \
+    or_else, coalesce, or_missing, binom_test, pchisqtail, pl_dosage, \
+    pl_to_gp, pnorm, pT, pF, ppois, qchisqtail, qnorm, qpois, range, zeros, \
+    rand_bool, rand_norm, rand_norm2d, rand_pois, rand_unif, rand_beta, \
+    rand_gamma, rand_cat, rand_dirichlet, sqrt, corr, str, is_snp, is_mnp, \
+    is_transition, is_transversion, is_insertion, is_deletion, is_indel, \
+    is_star, is_complex, is_strand_ambiguous, allele_type, hamming, \
+    mendel_error_code, triangle, downcode, gq_from_pl, parse_call, \
+    unphased_diploid_gt_index_call, argmax, argmin, zip, zip_with_index, map, \
+    flatmap, flatten, any, all, filter, sorted, find, group_by, fold, \
+    array_scan, len, min, nanmin, max, nanmax, mean, median, product, sum, \
+    cumulative_sum, struct, tuple, set, empty_set, array, empty_array, \
+    empty_dict, delimit, abs, sign, floor, ceil, float, float32, float64, \
+    parse_float, parse_float32, parse_float64, int, int32, int64, parse_int, \
+    parse_int32, parse_int64, bool, get_sequence, reverse_complement, \
+    is_valid_contig, is_valid_locus, contig_length, liftover, min_rep, \
+    uniroot, format, approx_equal, reversed, bit_and, bit_or, bit_xor, \
+    bit_lshift, bit_rshift, bit_not, binary_search, \
+    _values_similar, _showstr, _sort_by, _compare, _locus_windows_per_contig
```

Comment:
wow

Reply:
Hahha, yeah.  I'm not sure there's another alternative.  You can add things to all with:

```
__all__.extend(functions.__all__)
```

but you still need to get things into scope and if you want symbols to be checked, without `import *`.  I guess we could suppress the warning in this case, since we're importing from a module with restricted exports.  I'll do a bit more research to see how these things are handled.